### ISSUE TITLE
Column is incompatible with UserDefinedType

### ISSUE THREAD

**Issue by olirice** [sqlalchemy/sqlalchemy2-stubs#55]
**Describe the bug**
Columns with a `UserDefinedType` raise an error
```
error: Value of type variable "_TE" of "Column" cannot be <UserDefinedType>
```

**Expected behavior**
`UserDefinedTypes` are handled like builtin types.

**To Reproduce**
```py
from typing import Callable

from sqlalchemy import Column, Integer, MetaData
import sqlalchemy.types as types
from sqlalchemy.dialects.postgresql.base import PGDialect
from sqlalchemy.orm import declarative_base, Mapped

#########
# SETUP #
#########

class CITEXT(types.Concatenable, types.UserDefinedType):
    """Case Insensitive Text Type"""

    def literal_processor(self, dialect: PGDialect) -> Callable:
        def process(value: str) -> str:
            value = value.replace("'", "''")

            if dialect.identifier_preparer._double_percents:
                value = value.replace("%", "%%")

            return "'%s'" % value
        return process

    def get_col_spec(self) -> str:
        return "CITEXT"

    def bind_processor(self, dialect: PGDialect) -> Callable:
        def process(value: str) -> str:
            return value
        return process

    def result_processor(self, dialect: PGDialect, coltype: int) -> Callable:
        def process(value: str) -> str:
            return value
        return process

#########
# USAGE #
#########

Base = declarative_base(metadata=MetaData())

class User(Base):
    id = Column(Integer(), primary_key=True)
    email: Mapped[str] = Column(CITEXT) # error: Value of type variable "_TE" of "Column" cannot be "CITEXT" 

```

**Error**

```
# error: Value of type variable "_TE" of "Column" cannot be "CITEXT"
```

**Versions.**
 - OS: macOS 10.15.4
 - Python: 3.8.2
 - SQLAlchemy: 1.4.6
 - Database: all

**Context**

This appears to be happening because `Column` expects its type to be a subclass of `TypeEngine` 
https://github.com/sqlalchemy/sqlalchemy2-stubs/blob/4e1a045656a17eda57334c6bf3a147458ed0124c/sqlalchemy-stubs/sql/type_api.pyi#L41

but the stub for `UserDefinedType` is not
https://github.com/sqlalchemy/sqlalchemy2-stubs/blob/4e1a045656a17eda57334c6bf3a147458ed0124c/sqlalchemy-stubs/sql/type_api.pyi#L107

In contrast, [the definition in sqlalchemy proper](https://github.com/sqlalchemy/sqlalchemy/blob/ac12f663126a3977b49a9adca872be128a53b8d6/lib/sqlalchemy/sql/type_api.py#L714
) is
```py
class UserDefinedType(util.with_metaclass(VisitableCheckKWArg, TypeEngine)):
```

If its as simple as adding `TypeEngine` as a base I'd be happy to open a PR, but I'm guessing something to do with the metaclasses is a problem for mypy?

**Comment by bryanforbes** [comment]
I went ahead and fixed the inheritance of `UserDefinedType` in a PR (I'm not sure how I missed that). The fix for this will also include sqlalchemy/sqlalchemy#6223 to allow the plugin to properly infer the type from a subclass of `TypeDecorator` or `UserDefinedType`.

**Comment by olirice** [comment]
https://github.com/sqlalchemy/sqlalchemy/pull/6223 has landed and is in `sqlalchemy==1.4.8`

Confirmed using `1.4.8` + master of `sqlalchemy2-stubs` resolves the issue

This can close on the next release of `sqlalchemy2-stubs`. At time of writing, `0.0.1a4` is the current release.

**Comment by CaselIT** [comment]
1a5 was released, so this can be closed
